File: Example/Objc_Example/Channel Plugin Sample objc/UserViewController.m
//
//  LoginViewController.m
//  Channel Plugin Sample objc
//
//  Created by Haeun Chung on 21/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

#import "UserViewController.h"
#import <ChannelIO/ChannelIO-Swift.h>

@interface UserViewController () <UITextFieldDelegate>

@end

@implementation UserViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.pluginKeyField.placeholder = @"Insert your pluginKey";
    self.idField.placeholder = @"user Id";
    self.usernameField.placeholder = @"user name (optional)";
    self.phoneField.placeholder = @"mobile number (optional)";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [self.idField resignFirstResponder];
    [self.phoneField resignFirstResponder];
    [self.usernameField resignFirstResponder];
    return YES;
}

- (IBAction)didClickOnBoot:(id)sender {
    [self.idField resignFirstResponder];
    [self.phoneField resignFirstResponder];
    [self.usernameField resignFirstResponder];
    [self.pluginKeyField resignFirstResponder];
    
    Profile *profile = [[Profile alloc] init];
    [profile setWithMobileNumber:self.phoneField.text];
    [profile setWithName:self.usernameField.text];

    ChannelPluginSettings *settings = [[ChannelPluginSettings alloc] init];
    [settings setPluginKey:self.pluginKeyField.text];
    [settings setUserId:self.idField.text];

    [ChannelIO bootWith:settings profile:profile completion:^(ChannelPluginCompletionStatus status, Guest *guest) {

    }];
}

- (IBAction)didClickOnShutdown:(id)sender {
    [ChannelIO shutdown];
}

- (IBAction)didClickOnShowLauncher:(id)sender {
    [ChannelIO showWithAnimated:YES];
}

- (IBAction)didClickOnHideLauncher:(id)sender {
    [ChannelIO hideWithAnimated:YES];
}

- (IBAction)didClickOnOpenChat:(id)sender {
    [ChannelIO openWithAnimated:YES];
}
@end

File: Example/Swift Example/Channel Plugin Sample/UserViewController.swift
//
//  LoginViewController.swift
//  Example
//
//  Created by Haeun Chung on 14/03/2017.
//  Copyright © 2017 ZOYI. All rights reserved.
//

import Foundation
import UIKit
import ChannelIO

class UserViewController : UIViewController {
  @IBOutlet var idField: UITextField!
  @IBOutlet var usernameField: UITextField!
  @IBOutlet var phoneField: UITextField!
  @IBOutlet var loginButton: UIButton!
  @IBOutlet var pluginKeyField: UITextField!
  
  override func viewDidLoad() {
    super.viewDidLoad()
    self.pluginKeyField.placeholder = "Plugin Key"
    self.idField.placeholder = "user Id"
    self.usernameField.placeholder = "user name (optional)"
    self.phoneField.placeholder = "mobile number (optional)"
  }
  
  @IBAction func onClickBoot() {
    self.idField.resignFirstResponder()
    self.usernameField.resignFirstResponder()
    self.phoneField.resignFirstResponder()
    self.pluginKeyField.resignFirstResponder()
    
    guard var pluginKey = self.pluginKeyField.text else { return }
    let guest = Guest()
      .set(name: self.usernameField.text ?? "")
      .set(id: self.idField.text ?? "")
      .set(mobileNumber: self.phoneField.text ?? "")
    
    //pluginKey = "52eb6f27-38c7-476d-ad92-83e6299b7e07"
    let settings = ChannelPluginSettings(pluginKey: pluginKey)
    
    ChannelIO.boot(with: settings, guest: guest) { (completion) in
      switch completion {
      case .success:
        break
      default:
        break
      }
    }
  }
  
  @IBAction func onClickShutdown() {
    ChannelIO.shutdown()
  }
  
  @IBAction func onClickShowChat(_ sender: Any) {
    ChannelIO.open(animated: true)
  }
  
  @IBAction func onClickShowLauncher(_ sender: Any) {
    ChannelIO.show(animated: true)
  }
  
  @IBAction func onClickHideLauncher(_ sender: Any) {
    ChannelIO.hide(animated: true)
  }
}

extension UserViewController: UITextFieldDelegate {
  func textFieldShouldReturn(_ textField: UITextField) -> Bool {
    self.idField.resignFirstResponder()
    self.usernameField.resignFirstResponder()
    self.phoneField.resignFirstResponder()
    return true
  }
}
